% \VignetteIndexEntry{contrast Manual}
% \VignetteDepends{contrast}
% \VignetteKeywords{Sweave}
% \VignettePackage{contrast}
\documentclass[12pt]{article}
\usepackage{amsmath}
\usepackage[pdftex]{graphicx}
\usepackage{color}
\usepackage{xspace}
\usepackage{fancyvrb}
\usepackage{fancyhdr}
\usepackage{lastpage}
\usepackage{algorithm2e}
\usepackage[
         colorlinks=true,
         linkcolor=blue,
         citecolor=blue,
         urlcolor=blue]
         {hyperref}
\usepackage{Sweave}         
\usepackage{amsmath}
\usepackage{ctable}   

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% define new colors for use
\definecolor{darkgreen}{rgb}{0,0.6,0}
\definecolor{darkred}{rgb}{0.6,0.0,0}
\definecolor{lightbrown}{rgb}{1,0.9,0.8}
\definecolor{brown}{rgb}{0.6,0.3,0.3}
\definecolor{darkblue}{rgb}{0,0,0.8}
\definecolor{darkmagenta}{rgb}{0.5,0,0.5}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\newcommand{\contrast}{{\tt contrast}\xspace}
\newcommand{\contrastFunc}{{\tt contrast}\xspace}

\newcommand{\bld}[1]{\mbox{\boldmath $#1$}}
\newcommand{\shell}[1]{\mbox{$#1$}}
\renewcommand{\vec}[1]{\mbox{\bf {#1}}}
\newcommand{\grkbld}[1]{\mbox{\boldmath $#1$}}

\newcommand{\ReallySmallSpacing}{\renewcommand{\baselinestretch}{.6}\Large\normalsize}
\newcommand{\SmallSpacing}{\renewcommand{\baselinestretch}{1.1}\Large\normalsize}

\newcommand{\halfs}{\frac{1}{2}}

\setlength{\oddsidemargin}{-.25 truein}
\setlength{\evensidemargin}{0truein}
\setlength{\topmargin}{-0.2truein}
\setlength{\textwidth}{7 truein}
\setlength{\textheight}{8.5 truein}
\setlength{\parindent}{0.20truein}
\setlength{\parskip}{0.10truein}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\newcommand{\code}[1]{\mbox{\footnotesize\color{darkblue}\texttt{#1}}}
\newcommand{\pkg}[1]{{\textsf{#1}}}

\DefineVerbatimEnvironment{Sinput}{Verbatim}{fontshape=sl,formatcom=\color{darkblue}}
\fvset{listparameters={\setlength{\topsep}{0pt}}}
\renewenvironment{Schunk}{\vspace{\topsep}}{\vspace{\topsep}}

\fvset{fontsize=\footnotesize}

\SweaveOpts{keep.source=TRUE}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\pagestyle{fancy}
\lhead{}
\chead{The \texttt{contrast} Package}
\rhead{}
\lfoot{}
\cfoot{}
\rfoot{\thepage\ of \pageref{LastPage}}
\renewcommand{\headrulewidth}{1pt}
\renewcommand{\footrulewidth}{1pt}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\title{The \texttt{contrast} Package}
\author{Max Kuhn, Steve Weston, Jed Wing, James Forester}

<<loadLib, results = hide, echo = FALSE>>=
library(contrast)
library(MASS)
library(nlme)
library(lattice)
library(Hmisc)
library(geepack)
options(useFancyQuotes = FALSE, width = 80)
@

\begin{document}
\SweaveOpts{concordance=TRUE}

\maketitle

\thispagestyle{empty}
	
\section{Introduction}

The purpose of the \pkg{contrast} package is to provide a standardized interface for testing linear combinations of parameters from common regression models. The syntax mimics the \code{contrast.Design} function from the \pkg{Design} library. The contrast class has been extended in this package to linear models produced using the functions \code{lm}, \code{glm}, \code{gls}, \code{lme} and \code{geese}. Other \code{R} functions with similar purposes exist in \pkg{R}, but the interfaces are different and many require the user to specify the contrast in terms of the parameter contrast coefficient vector. This package aims to simplify the process for the user.

\section{Contrasts}

First, some notation:

\begin{align}
   n &= \text{number of samples} \notag \\
   p &= \text{number of model parameters associated with fixed effects (excluding the intercept)} \notag \\
   q &= \text{number of covariance parameters with random effects or correlations } \notag \\
   Y &= \text{$n\times 1$ response vector} \notag \\
   X &= \text{$n\times (p+1)$ model matrix} \notag \\
   \beta &= \text{model parameters associated with fixed effects} \notag \\
   \Sigma &= \text{covariance matrix associated with the fixed effects} \notag \\
\end{align}

This package uses one degree of freedom Wald tests to calculate p--values for linear combinations of parameters. For example, the basic linear model is of the form $y=X\beta+\epsilon$, where the individual errors are assumed to be iid $N(0, \sigma^2)$. Ordinary least squares provides us with estimates $\hat{\beta}$, $\hat{\sigma}^2$ and $\hat{\Sigma}$. Given a $(p+1)\times 1$ vector of constants, $c$, we can estimate a linear combination of parameters $\lambda = c'\beta$ by substituting the estimated parameter vectors: $\hat{\lambda} = c'\hat{\beta}$. Using basic linear algebra, $Var[\lambda] =  c'\Sigma c$. The statistic generated for contrasts is
\begin{equation}\label{E:Wald}
S = \frac{c'\hat{\beta}}{\sqrt{c'\hat{\Sigma} c}} 
\end{equation}
For linear models with normal errors, $S\sim T_{n-p-1}$ and there is no uncertainty about the distribution of the test statistic and the degrees of freedom. In other cases, this is not true. Asymptotics come into play for several models and there is some ambiguity as to whether a $t$ or normal distribution should be used to compute p--values (See Harrell, 2001, Section 9,2 for a discussion). We follow the conventions of each package: \code{glm}, \code{gls} and \code{lme} models use a $t$ distribution and a normal distribution is used for \code{gee} models. For models where there are extra covariance or correlation parameters, we again follow the lead of the package. For \code{gls} model, the degrees of freedom are $n-p$, while in \code{lme} models, it is $n-p-q$.

The remainder of this document shows two examples and how the \code{contrast} function can be applied to different models.

\subsection{Linear Models}
 


As an example, a gene expression experiment was run to assess the effect of a compound under two different diets: high fat and low fat. The main comparisons of interest are the difference between the treated and untreated groups within a diet. The interaction effect was a secondary hypothesis. For illustration, we only include the expression value of one of the genes.

A summary of the design is given in Table \ref{designSummary1}.

<<example1Data, echo = FALSE, results = hide>>=
example1 <- expand.grid(
   tissue = c("Whole Blood", "PBMC"),
   trmt = c("Vehicle", "Compound1", "Compound2"),   
   rep = 1:3)
example1 <- data.frame(
   expression = 
   c(8.029, 7.650, 7.304, 7.240, 7.322, 7.727, 
     7.554, 7.198, 7.575, 7.399, 8.211, 7.321,
     7.548, 7.538, 7.565, 7.304, 7.998, 8.067, 
     7.755, 7.415, 7.850, 7.217, 8.030, 7.432),
   diet =
   c("high fat",  "high fat",  "high fat",   "low fat",  
      "low fat",   "low fat",   "low fat",   "low fat", 
      "low fat",   "low fat",  "high fat",   "low fat",  
      "low fat",   "low fat",   "low fat",   "low fat", 
     "high fat",  "high fat",  "high fat",  "high fat",  
     "high fat",  "high fat",  "high fat",  "high fat"),
   group =  
   c("treatment",     "vehicle",     "vehicle",  "vehicle",   
      "vehicle",   "treatment",   "treatment",  "vehicle",
      "treatment",    "vehicle",   "treatment",  "vehicle",    
      "treatment",  "treatment",   "treatment",  "vehicle",    
      "treatment",  "treatment",     "vehicle",  "vehicle",    
      "treatment",    "vehicle",   "treatment",  "vehicle"))
      
   designSummary1 <- as.data.frame(table(example1$diet, example1$group))
   names(designSummary1)[1:2] <- c("Diet", "Group")
@
 
<<designSummary1, echo = FALSE, results = tex>>=
latex(
      designSummary1, 
      ctable = TRUE, 
      rowname=NULL, 
      caption = "A summary of the diet experimental design",
      where = "h",
      file = "")
@

The study design was a two--way factorial with $n=24$. The cell means can be labeled as in Table \ref{design1Cells}.

\setkeys{Gin}{width=.6\textwidth}
\begin{figure}[t]
   \begin{center}		
<<example1Plot, fig = TRUE, echo = FALSE, width = 6, height = 4.25>>=
plotTheme <- col.whitebg()
plotTheme$superpose.line$lty <- rep(1, 8)

trellis.par.set(plotTheme, warn = FALSE)

out <- stripplot(
                 expression ~ group, 
                 example1,  
                 groups = diet,
                 as.Table = TRUE,
                 type = c("p", "a", "g"),
                 auto.key = list(columns = 2),
                 ylab = "Expression")  
print(out)

@
      \caption{An interaction plot for the diet gene expression experiment}
      \label{f:exp1}         
   \end{center}
\end{figure}

\ctable[label=design1Cells, pos=!h, caption = {The cells of the diet experimental design}]{rcc} 
 {} 
 {
& \multicolumn{2}{c}{ Diet}\NN
& Low Fat & High Fat \LL
Vehicle			& $A$		& $B$ 	\NN
Compound		& $C$		& $D$ 	\LL
}	 

The reference cell used by {\tt R}  is cell $D$, the treated samples on a high fat diet.

The model used is
\begin{align}
\log\text{Expression}_2 &= \beta_0  \notag \\
	& + \beta_1\text{Vehicle Group} \notag \\
	& + \beta_2\text{Low Fat Diet} \notag \\
	& + \beta_{3}\text{Low Fat Diet and Vehicle Group} \label{E:model1}
\end{align}
so that $p=3$. Substituting the appropriate coefficients into each cell produces the parameters in Table \ref{design1Coef}.
\ctable[label=design1Coef, pos=!h, caption = {The parameter structure of the diet experimental design} ]{rcc} 
 {} 
 {
& \multicolumn{2}{c}{Diet}\NN
& Low Fat & High Fat \LL
Vehicle			& $\beta_0 + \beta_1 + \beta_2 + \beta_{3}$		& $\beta_0 +  \beta_1 $ 	\NN
Compound		& $\beta_0 + \beta_2$		& $\beta_0$ 	\LL
}	 

This means that
\begin{itemize}
	\item $\beta_2$ tests for diet effect in the treated samples ($C-D$)
	\item $\beta_1$ tests for a compounds effect in the high fat diet samples ($B-D$)
\end{itemize}


Fitting the model specified by  (\ref{E:model1}) using \code{lm}:
<<example1LinearMod>>=
lmFit1 <- lm(expression ~ (group + diet)^2, data = example1)
summary(lmFit1)
@


To test the treatment effect in the high fat diet, $D-B = -\beta_1$. This coefficient and hypothesis test for the difference between treated and un--treated in the high fat diet group is in the row labeled as \code{groupvehicle} in the output of \code{summary.lm}. 


To compare the compound data and the vehicle data in the low fat diet group, Tables \ref{design1Cells} and \ref{design1Coef} can be used:
\begin{align}
C - A &= 	\beta_0 + \beta_2 -(\beta_0+\beta_1+\beta_2+\beta_{3}) \notag \\
	&= -\beta_1 - \beta_{3} \notag
\end{align}
This hypothesis translates to testing $\beta_1 + \beta_{3} = 0$, or a contrast using $c=(0, 1, 0, 1)$. To get the results of the difference between treated and un--treated in the low fat diet group, we (finally) use the contrast function:
<<example1Contrast>>=
highFatDiff <- contrast(lmFit1, 
                        list(diet = "low fat", group = "vehicle"),
                        list(diet = "low fat", group = "treatment"))
print(highFatDiff, X = TRUE)
@

<<example1ContrastStat, echo = FALSE, results = hide>>=
basicTestStat <- highFatDiff$testStat
@   
While the effect of treatment is significantly different when compared to vehicle for both diets, the difference is more pronounced in the high fat diet.

Alternatively, both test can be done in the same call to \code{contrast}:
<<eachTest>>=
eachTrmtEffect <- contrast(lmFit1, 
                           list(diet = levels(example1$diet), group = "vehicle"),
                           list(diet = levels(example1$diet), group = "treatment"))  
print(eachTrmtEffect, X = TRUE)
@ 


Also, we can use the \code{type} argument to compute a single treatment effect averaging over the levels of the other factor:
<<meanEffect>>=
meanTrmtEffect <- contrast(lmFit1, 
                           list(diet = levels(example1$diet), group = "vehicle"),
                           list(diet = levels(example1$diet), group = "treatment"),
                           type = "average")  
  
print(meanTrmtEffect, X = TRUE)
@ 

 
Additionally, for ordinary linear regression models, there is an option to use sandwich estimates for the covariance matrix of the parameters. See the \href{http://cran.r-project.org/src/contrib/Descriptions/sandwich.html}{sandwich} package for more details. Going back to our comparison of treated versus control in low fat samples, we can use the HC3 estimate in the contrast. 
<<example1Sand>>=
highFatDiffSAND <- contrast(lmFit1, 
                            list(diet = "low fat", group = "vehicle"),
                            list(diet = "low fat", group = "treatment"),
                            covType = "HC3")
print(highFatDiffSAND)
@


The $t$--statistic associated with the sandwich estimate is \Sexpr{round(highFatDiffSAND$testStat, 3)} versus \Sexpr{round(basicTestStat, 3)} using the traditional estimate of the covariance matrix.


<<lmTest, results = hide, echo = FALSE>>=
lmTest <- contrast(lmFit1, 
                   list(diet = "high fat", group = "vehicle"),
                   list(diet = "high fat", group = "treatment"))
if(abs(lmTest$Pvalue- summary(lmFit1)$coefficients[2,4]) > 0.0001)
stop("lm test case failed")      
@

\subsection{Generalized Linear Model}

In this class of models, the distributional assumptions are expanded beyond the normal distribution to the general exponential family. Also, these models are linear in the sense that they are linear on a specified scale. The link function, denoted as $\eta$, is a function that defines how the linear predictor, $x'\beta$, enters the model. While there are several approaches to testing for statistical differences between models, such as the likelihood ratio or score tests, the Wald test is another method for assessing the statistical significance of linear combinations of model parameters. The basic Wald--type test uses the familiar statistic \ref{E:Wald} to evaluate hypotheses. The distributional properties are exact for the normal distribution and asymptotically valid for other distributions in the exponential family. There are some issues with the Wald test (see Hauck and Donner, 1977). Whenever possible, likelihood ratio or score statistics are preferred, but these tests cannot handle some types of hypotheses, in which case the Wald test can be used.

For the previous example, it is customary to log transform gene expression data using a base of 2, we can illustrate contrasts in generalized linear models using the log (base $e$) link. In this case, the actual model being fit is $\exp(x'\beta)$.  
<<example1GenLinearMod>>=
glmFit1 <- glm(2^expression ~ (group + diet)^2, 
               data = example1, 
               family = gaussian(link = "log"))
summary(glmFit1)
highFatDiff <- contrast(glmFit1, 
                        list(diet = "low fat", group = "vehicle"),
                        list(diet = "low fat", group = "treatment"))
print(highFatDiff, X = TRUE)
@


The coefficients and p--values are not wildly different given that the scale is slightly different (i.e. log$_2$ versus log$_e$).

<<glmTest, results = hide, echo = FALSE>>=
glmTest <- contrast(glmFit1, 
                    list(diet = "high fat", group = "vehicle"),
                    list(diet = "high fat", group = "treatment"))
if(abs(glmTest$Pvalue - summary(glmFit1)$coefficients[2,4]) > 0.0001)
  stop("glm test case failed")       
@


\subsection{Generalized Least Squares}

In a second gene expression example, stem cells were differentiated using a set of factors (such as media types, cell spreads etc.). These factors were collapsed into a single cell environment configurations variable. The cell lines were assays over three days. Two of the configurations were only run on the first day and the other two were assays at baseline. 

To get the materials, three donors provided materials. These donors provided (almost) equal replication across the two experimental factors (day and configuration). Table \ref{example2Table} shows a summary of the design.

<<example2Data, echo = FALSE, results = hide>>=
 example2 <- data.frame(
   subject =
      c("donor1", "donor1", "donor1", "donor1", "donor1", "donor1", "donor1", 
        "donor2", "donor2", "donor2", "donor3", "donor2", "donor2", "donor2", 
        "donor2", "donor3", "donor3", "donor3", "donor3", "donor3", "donor3", 
        "donor3", "donor1"),
   config = 
      c("A", "D", "C", "D", "C", "D", "C", 
        "B", "A", "D", "B", "C", "D", "D", 
        "C", "A", "D", "C", "D", "D", "C", 
        "C", "B"),
   day = c(1, 1, 1, 2, 2, 4, 4, 1, 1, 1, 1, 1, 2, 4, 4, 1, 1, 1, 2, 4, 2, 4, 1),
   expression = 
      c(9.364, 9.533, 9.267, 9.407, 9.463, 9.419, 9.507, 
        9.226, 9.153, 9.112, 9.657, 8.973, 9.096, 9.232, 
        9.261, 9.383, 9.365, 9.242, 9.361, 9.345, 9.444, 
        9.551, 9.601))
   example2Table <- as.data.frame(table(example2$day, example2$config))
   names(example2Table) <- c("Day", "Configuration", "Number of Donors")
@

<<designSummary2, echo = FALSE, results = tex>>=
latex(
      example2Table, 
      ctable = TRUE, 
      rowname=NULL, 
      caption = "A summary of the stem cell experimental design",
      where = "h",
      file = "")
@


The one of the goal of this experiment was to assess pre--specified differences in the configuration at each time point. For example, the differences between configurations A and B at day one is of interest. Also, the differences between configurations C and D at each time points were important.

Since there are missing cells in the design, it is not a complete two--way factorial. One way to analyze this experiment is to further collapse the time and configuration data into a single variable and then specify each comparison using this factor.

For example:
<<design2factor>>=
example2$group <- factor(paste(example2$day, ":", example2$config, sep = ""))
print(table(example2$group))
@

Using this new factor, we fit a linear model to this one--way design. We should account for the possible within--donor correlation. A generalized least square fit can do this, where we specify a correlation structure for the residuals. A compound--symmetry (a.k.a. exchangeable) correlation structure assumes that the within--donor correlation is constant. 

The mdoel fit is:
<<design2gls>>=
glsFit <-  gls(expression ~ group, 
               data = example2, 
               corCompSymm(form = ~ 1 | subject))
summary(glsFit)
@

In this example, $n=23$ and $p=8$. This model estimates the residual variance and the within--subject correlation, so $q=2$. The default parameter estimates compare each group to the reference cell (day 1, configuration A). The summary table provides one of the p--values that we are interested in (configuration A vs. B at day 1). An example of obtaining the other p--values is shown below:
<<design2glsCont>>=
print(
      contrast(glsFit, 
               list(group = "4:C"),
               list(group = "4:D")),
      X = TRUE)     
@


\setkeys{Gin}{width=.6\textwidth}
\begin{figure}[t]
   \begin{center}		
<<example2Plot, fig = TRUE, echo = FALSE, width = 6, height = 4.25>>=
out2 <- stripplot(
                  expression ~ factor(day), 
                  example2,  
                  groups = factor(config),
                  as.Table = TRUE,
                  type = c("p", "a", "g"),
                  xlab = "Day",
                  ylab = "Log2 Expression",
                  auto.key = list(columns = 4))   
print(out2)
@
      \caption{An interaction plot for the stem cell gene expression experiment}
      \label{f:exp2}         
   \end{center}
\end{figure}
      

<<glsTest, results = hide, echo = FALSE>>=
glsTest <- contrast(glsFit, 
                    list(group = "1:A"),
                    list(group = "4:D"))
if(abs(glsTest$Pvalue - summary(glsFit)$tTable[8,4])  > 0.0001)
  stop("gls test case failed")      
@

\subsection{Linear Mixed Models via \code{lme}}

A similar model can be fit using a linear mixed model via the \code{lme} function. In this case, we can add a random intercept attributable to the donors. This can produce the above compound symmetry model, but here the within donor--correlation is constrained to be positive.
<<design2lme>>=
lmeFit <-  lme(expression ~ group, 
               data = example2, 
               random = ~1|subject)
summary(lmeFit)

print(
      contrast(lmeFit, 
               list(group = "4:C"),
               list(group = "4:D")),
      X = TRUE)        
@

Comparing this to the \code{gls} model results, the default coefficients have identical parameter estimates, standard errors and test statistics, but their $p$--values are slightly different. This is due to the difference in how the degrees of freedom are calculated between these models. The same is true for the example contrast for the two models (15 versus 13 degrees of freedom).


<<lmeTest, results = hide, echo = FALSE>>=
lmeTest <- contrast(lmeFit, 
                    list(group = "1:A"),
                    list(group = "4:D"))
if(abs(lmeTest$Pvalue - summary(lmeFit)$tTable[8,5])  > 0.0001)
  stop("lme test case failed")      
@
\subsection{Generalized Estimating Equations}      
      
Yet another way to fit a model to these data would be to use a generalized linear model--type framework using normal errors and a log (base 2) link. To account for the within--donor variability, a generalized estimating equation approach can be used. We use the \code{geese} function in the \code{geepack} package.
<<design2lme>>=
geeFit <-  geese(2^expression ~ group,
                 data = example2,
                 id = subject,
                 family = gaussian(link = "log"),
                 corstr = "exchangeable")
summary(geeFit)

print(
      contrast(geeFit, 
               list(group = "4:C"),
               list(group = "4:D")),
      X = TRUE)   
@
For this model, a simple Wald test is calculated. The contrast shows a more significant p--value than the other models, partly due to the scale and partly due to the distributional assumptions about the test statistic.

<<geeTest, results = hide, echo = FALSE>>=
geeTest <- contrast(geeFit, 
                    list(group = "1:A"),
                    list(group = "4:D"))
if(abs(geeTest$Pvalue - summary(geeFit)$mean[8,4])  > 0.0001)
  stop("gee test case failed")      
@     

\section{Fold changes}

The \code{contrast} method also computes fold changes using the follow process:

\begin{enumerate}
\item For the two groups defined by the \code{a} and \code{b} arguments, the predicted outcomes are computed. When the model objects is generated by either \code{glm} or \code{geese}, the linear predictor is calculated.
\item The two predicted values are optionally transformed by the \code{fcFunc} argument. For our gene expression example, we might use \verb+function(u) 2^u+, while for generalized linear models we might use the inverse link function from the \code{family} object. 
\item The predicted value for the \code{a} group is divided by the predicted value for the \code{b} group.
\item If the \code{fcType} argument is \code{"simple"}, the ratio is returned as the fold change. \code{fcType = "log"} returns the log (base $e$) of the ratio. If the type is \code{"signed"}, a different calculation is used. If the simple ratio is less than one, the negative reciprocal of the ratio is returned; otherwise the fold change is equal to the simple ratio
\end{enumerate}

The fold change results are contained in the output as \code{foldchange}. From the first example:
<<ex1FC>>=
eachTrmtEffect <- contrast(lmFit1, 
                           list(diet = levels(example1$diet), group = "vehicle"),
                           list(diet = levels(example1$diet), group = "treatment"),
                           fcfunc = function(u)2^u)  
print(eachTrmtEffect, X = TRUE)
eachTrmtEffect$foldChange
@ 


\section{References}


\begin{description}

\item [] Harrell, F. E. (2001), {\em Regression Modeling Strategies}. New York: Springer-Verlag.

\item [] Hauck, W. W. and Donner, A.  (1977), Wald's Test as Applied to Hypotheses in Logit Analysis. {\em Journal of the American Statistical Association}
{\bf 72}, 851-863.

\end{description}
\end{document}

