<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: N-integration with Projection to Latent Structures models...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for block.plsda {mixOmics}"><tr><td>block.plsda {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>N-integration with Projection to Latent Structures models (PLS) with
Discriminant Analysis</h2>

<h3>Description</h3>

<p>Integration of multiple data sets measured on the same samples or
observations to classify a discrete outcome, ie. N-integration with
Discriminant Analysis. The method is partly based on Generalised Canonical
Correlation Analysis.
</p>


<h3>Usage</h3>

<pre>
block.plsda(
  X,
  Y,
  indY,
  ncomp = 2,
  design,
  scheme,
  scale = TRUE,
  init = "svd",
  tol = 1e-06,
  max.iter = 100,
  near.zero.var = FALSE,
  all.outputs = TRUE
)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>A named list of data sets (called 'blocks') measured on the same 
samples. Data in the list should be arranged in matrices, samples x variables, 
with samples order matching in all data sets.</p>
</td></tr>
<tr valign="top"><td><code>Y</code></td>
<td>
<p>a factor or a class vector for the discrete outcome.</p>
</td></tr>
<tr valign="top"><td><code>indY</code></td>
<td>
<p>To supply if <code>Y</code> is missing, indicates the position of 
the matrix response in the list <code>X</code>.</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>the number of components to include in the model. Default to 2.
Applies to all blocks.</p>
</td></tr>
<tr valign="top"><td><code>design</code></td>
<td>
<p>numeric matrix of size (number of blocks in X) x (number of
blocks in X) with values between 0 and 1. Each value indicates the strenght
of the relationship to be modelled between two blocks; a value of 0
indicates no relationship, 1 is the maximum value. Alternatively, one of
c('null', 'full') indicating a disconnected or fully connected design,
respecively, or a numeric between 0 and 1 which will designate all
off-diagonal elements of a fully connected design (see examples in
<code>block.splsda</code>). If <code>Y</code> is provided instead of <code>indY</code>, the
<code>design</code> matrix is changed to include relationships to <code>Y</code>.</p>
</td></tr>
<tr valign="top"><td><code>scheme</code></td>
<td>
<p>Character, one of 'horst', 'factorial' or 'centroid'. Default =
<code>'horst'</code>, see reference.</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>Logical. If scale = TRUE, each block is standardized to zero
means and unit variances (default: TRUE)</p>
</td></tr>
<tr valign="top"><td><code>init</code></td>
<td>
<p>Mode of initialization use in the algorithm, either by Singular
Value Decomposition of the product of each block of X with Y ('svd') or each
block independently ('svd.single'). Default = <code>svd.single</code>.</p>
</td></tr>
<tr valign="top"><td><code>tol</code></td>
<td>
<p>Positive numeric used as convergence criteria/tolerance during the
iterative process. Default to <code>1e-06</code>.</p>
</td></tr>
<tr valign="top"><td><code>max.iter</code></td>
<td>
<p>Integer, the maximum number of iterations. Default to  100.</p>
</td></tr>
<tr valign="top"><td><code>near.zero.var</code></td>
<td>
<p>Logical, see the internal <code><a href="../../mixOmics/help/nearZeroVar.html">nearZeroVar</a></code>
function (should be set to TRUE in particular for data with many zero
values). Setting this argument to FALSE (when appropriate) will speed up the
computations. Default value is FALSE.</p>
</td></tr>
<tr valign="top"><td><code>all.outputs</code></td>
<td>
<p>Logical. Computation can be faster when some specific
(and non-essential) outputs are not calculated. Default = <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>block.plsda</code> function fits a horizontal integration PLS-DA model with
a specified number of components per block). A factor indicating the
discrete outcome needs to be provided, either by <code>Y</code> or by its position
<code>indY</code> in the list of blocks <code>X</code>.
</p>
<p><code>X</code> can contain missing values. Missing values are handled by being
disregarded during the cross product computations in the algorithm
<code>block.pls</code> without having to delete rows with missing data.
Alternatively, missing data can be imputed prior using the
<code><a href="../../mixOmics/help/impute.nipals.html">impute.nipals</a></code> function.
</p>
<p>The type of algorithm to use is specified with the <code>mode</code> argument.
Four PLS algorithms are available: PLS regression <code>("regression")</code>, PLS
canonical analysis <code>("canonical")</code>, redundancy analysis
<code>("invariant")</code> and the classical PLS algorithm <code>("classic")</code> (see
References and <code>?pls</code> for more details).
</p>
<p>Note that our method is partly based on Generalised Canonical Correlation
Analysis and differs from the MB-PLS approaches proposed by Kowalski et al.,
1989, J Chemom 3(1) and Westerhuis et al., 1998, J Chemom, 12(5).
</p>


<h3>Value</h3>

<p><code>block.plsda</code> returns an object of class
<code>"block.plsda","block.pls"</code>, a list that contains the following
components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>the centered and standardized original predictor matrix.</p>
</td></tr>
<tr valign="top"><td><code>indY</code></td>
<td>
<p>the position of the outcome Y in the output list X.</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>the number of components included in the model for each block.</p>
</td></tr>
<tr valign="top"><td><code>mode</code></td>
<td>
<p>the algorithm used to fit the model.</p>
</td></tr> <tr valign="top"><td><code>variates</code></td>
<td>
<p>list
containing the variates of each block of X.</p>
</td></tr> <tr valign="top"><td><code>loadings</code></td>
<td>
<p>list containing
the estimated loadings for the variates.</p>
</td></tr> <tr valign="top"><td><code>names</code></td>
<td>
<p>list containing the
names to be used for individuals and variables.</p>
</td></tr> <tr valign="top"><td><code>nzv</code></td>
<td>
<p>list containing
the zero- or near-zero predictors information.</p>
</td></tr> <tr valign="top"><td><code>iter</code></td>
<td>
<p>Number of
iterations of the algorithm for each component</p>
</td></tr>
<tr valign="top"><td><code>prop_expl_var</code></td>
<td>
<p>Percentage of explained variance for each
component and each block</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Florian Rohart, Benoit Gautier, Kim-Anh Lê Cao, Al J Abadi
</p>


<h3>References</h3>

<p>On PLSDA:
</p>
<p>Barker M and Rayens W (2003). Partial least squares for discrimination.
<em>Journal of Chemometrics</em> <b>17</b>(3), 166-173. Perez-Enciso, M. and
Tenenhaus, M. (2003). Prediction of clinical outcome with microarray data: a
partial least squares discriminant analysis (PLS-DA) approach. <em>Human
Genetics</em> <b>112</b>, 581-592. Nguyen, D. V. and Rocke, D. M. (2002). Tumor
classification by partial least squares using microarray gene expression
data. <em>Bioinformatics</em> <b>18</b>, 39-50.
</p>
<p>On multiple integration with PLS-DA: Gunther O., Shin H., Ng R. T. ,
McMaster W. R., McManus B. M. , Keown P. A. , Tebbutt S.J. , Lê Cao K-A. ,
(2014) Novel multivariate methods for integration of genomics and proteomics
data: Applications in a kidney transplant rejection study, OMICS: A journal
of integrative biology, 18(11), 682-95.
</p>
<p>On multiple integration with sPLS-DA and 4 data blocks:
</p>
<p>Singh A., Gautier B., Shannon C., Vacher M., Rohart F., Tebbutt S. and Lê
Cao K.A. (2016). DIABLO: multi omics integration for biomarker discovery.
BioRxiv available here:
<a href="http://biorxiv.org/content/early/2016/08/03/067611">http://biorxiv.org/content/early/2016/08/03/067611</a>
</p>
<p>mixOmics article:
</p>
<p>Rohart F, Gautier B, Singh A, Lê Cao K-A. mixOmics: an R package for 'omics
feature selection and multiple data integration. PLoS Comput Biol 13(11):
e1005752
</p>


<h3>See Also</h3>

<p><code><a href="../../mixOmics/help/plotIndiv.html">plotIndiv</a></code>, <code><a href="../../mixOmics/help/plotArrow.html">plotArrow</a></code>,
<code><a href="../../mixOmics/help/plotLoadings.html">plotLoadings</a></code>, <code><a href="../../mixOmics/help/plotVar.html">plotVar</a></code>, <code><a href="../../mixOmics/help/predict.html">predict</a></code>,
<code><a href="../../mixOmics/help/perf.html">perf</a></code>, <code><a href="../../mixOmics/help/selectVar.html">selectVar</a></code>, <code><a href="../../mixOmics/help/block.pls.html">block.pls</a></code>,
<code><a href="../../mixOmics/help/block.splsda.html">block.splsda</a></code> and http://www.mixOmics.org for more details.
</p>


<h3>Examples</h3>

<pre>
data(nutrimouse)
data = list(gene = nutrimouse$gene, lipid = nutrimouse$lipid, Y = nutrimouse$diet)
# with this design, all blocks are connected
design = matrix(c(0,1,1,1,0,1,1,1,0), ncol = 3, nrow = 3,
byrow = TRUE, dimnames = list(names(data), names(data)))

res = block.plsda(X = data, indY = 3) # indY indicates where the outcome Y is in the list X
plotIndiv(res, ind.names = FALSE, legend = TRUE)
plotVar(res)

## Not run: 
# when Y is provided
res2 = block.plsda(list(gene = nutrimouse$gene, lipid = nutrimouse$lipid),
Y = nutrimouse$diet, ncomp = 2)
plotIndiv(res2)
plotVar(res2)

## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.18.1 <a href="00Index.html">Index</a>]</div>
</div></body></html>
