<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Clustered Image Maps (CIMs) (&quot;heat maps&quot;)</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body>

<table width="100%" summary="page for cim {mixOmics}"><tr><td>cim {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Clustered Image Maps (CIMs) (&quot;heat maps&quot;)</h2>

<h3>Description</h3>

<p>This function generates color-coded Clustered Image Maps (CIMs) (&quot;heat maps&quot;) to
represent &quot;high-dimensional&quot; data sets.
</p>


<h3>Usage</h3>

<pre>
cim(mat,
color = NULL,
row.names = TRUE,
col.names = TRUE,
row.sideColors = NULL,
col.sideColors = NULL,
row.cex = NULL,
col.cex = NULL,
threshold = 0,
cluster = "both",
dist.method = c("euclidean", "euclidean"),
clust.method = c("complete", "complete"),
cut.tree = c(0, 0),
transpose = FALSE,
symkey = TRUE,
keysize = c(1, 1),
keysize.label = 1,
zoom = FALSE,
title = NULL,
xlab = NULL,
ylab = NULL,
margins = c(5, 5),
lhei = NULL,
lwid = NULL,
comp=NULL,
center = TRUE,
scale = FALSE,
mapping = "XY",
legend= NULL,
save = NULL,
name.save = NULL)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>mat</code></td>
<td>
<p>numeric matrix of values to be plotted. Alternatively, an object of class inheriting from <code>"pca"</code>, <code>"spca"</code>, <code>"ipca"</code>, <code>"sipca"</code>,   <code>"rcc"</code>, <code>"pls"</code>, <code>"spls"</code>, <code>"plsda"</code>, <code>"splsda"</code>, <code>"mlspls"</code> or <code>"mlsplsda"</code> (where <code>"ml"</code> stands for multilevel).</p>
</td></tr>
<tr valign="top"><td><code>color</code></td>
<td>
<p>a character vector of colors such as that generated by <code><a href="../../grDevices/html/palettes.html">terrain.colors</a></code>,
<code><a href="../../grDevices/html/palettes.html">topo.colors</a></code>, <code><a href="../../grDevices/html/palettes.html">rainbow</a></code>, <code><a href="colors.html">color.jet</a></code> or similar functions.</p>
</td></tr>
<tr valign="top"><td><code>row.names, col.names</code></td>
<td>
<p>logical, should the name of rows and/or columns of <code>mat</code> be shown? If <code>TRUE</code>
(defaults) <code>rownames(mat)</code> and/or <code>colnames(mat)</code> are used. Possible character vectors with
row and/or column labels can be used.</p>
</td></tr>
<tr valign="top"><td><code>row.sideColors</code></td>
<td>
<p>(optional) character vector of length <code>nrow(mat)</code> containing the color
names for a vertical side bar that may be used to annotate the rows of <code>mat</code>.</p>
</td></tr>
<tr valign="top"><td><code>col.sideColors</code></td>
<td>
<p>(optional) character vector of length <code>ncol(mat)</code> containing
the color names for a horizontal side bar that may be used to annotate the columns of <code>mat</code>.</p>
</td></tr>
<tr valign="top"><td><code>row.cex, col.cex</code></td>
<td>
<p>positive numbers, used as <code>cex.axis</code> in for the row or column
axis labeling. The defaults currently only use number of rows or columns, respectively.</p>
</td></tr>
<tr valign="top"><td><code>mapping</code></td>
<td>
<p>character string indicating whether to map <code>"X"</code>,
<code>"Y"</code> or <code>"XY"</code>-association matrix. See Details.</p>
</td></tr>
<tr valign="top"><td><code>cluster</code></td>
<td>
<p>character string indicating whether to cluster <code>"none"</code>, <code>"row"</code>,
<code>"column"</code> or <code>"both"</code>. Defaults to <code>"both"</code>.</p>
</td></tr>
<tr valign="top"><td><code>dist.method</code></td>
<td>
<p>character vector of length two. The distance measure used in
clustering rows and columns. Possible values are <code>"correlation"</code> for Pearson
correlation and all the distances supported by <code><a href="../../stats/html/dist.html">dist</a></code>, such as
<code>"euclidean"</code>, etc.</p>
</td></tr>
<tr valign="top"><td><code>clust.method</code></td>
<td>
<p>character vector of length two. The agglomeration method to be used for rows and columns.
Accepts the same values as in <code><a href="../../stats/html/hclust.html">hclust</a></code> such as <code>"ward"</code>, <code>"complete"</code>, etc.</p>
</td></tr>
<tr valign="top"><td><code>cut.tree</code></td>
<td>
<p>numeric vector of length two with components in [0,1]. The height proportions where
the trees should be cut for rows and columns, if these are clustered.</p>
</td></tr>
<tr valign="top"><td><code>comp</code></td>
<td>
<p>atomic or vector of positive integers. The components to adequately account
for the data association. For a non sparse method, the similarity matrix is computed based on the variates and loading vectors of those specified components. For a sparse approach, the similarity matric is computed based on the variables selected on those specified components. See example. Defaults to <code>comp = 1:object$ncomp</code>.</p>
</td></tr>
<tr valign="top"><td><code>transpose</code></td>
<td>
<p>logical indicating if the matrix should be transposed for plotting.
Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr valign="top"><td><code>center</code></td>
<td>
<p>either a logical value or a numeric vector of length equal to the
number of columns of <code>mat</code>. See <code><a href="../../base/html/scale.html">scale</a></code> function.</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>either a logical value or a numeric vector of length equal to the
number of columns of <code>mat</code>. See <code><a href="../../base/html/scale.html">scale</a></code> function.</p>
</td></tr>
<tr valign="top"><td><code>threshold</code></td>
<td>
<p>numeric between 0 and 1. Variables with correlations below this threshold in absolute value are not plotted. To use only when mapping is &quot;XY&quot;.</p>
</td></tr>
<tr valign="top"><td><code>symkey</code></td>
<td>
<p>boolean indicating whether the color key should be made symmetric about 0.
Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr valign="top"><td><code>keysize</code></td>
<td>
<p>vector of length two, indicating the size of the color key.</p>
</td></tr>
<tr valign="top"><td><code>keysize.label</code></td>
<td>
<p>vector of length 1, indicating the size of the labels and title of the color key.</p>
</td></tr>
<tr valign="top"><td><code>zoom</code></td>
<td>
<p>logical. Whether to use zoom for interactive zoom. See Details.</p>
</td></tr>
<tr valign="top"><td><code>title, xlab, ylab</code></td>
<td>
<p>title, <i>x</i>- and <i>y</i>-axis titles; default to none.</p>
</td></tr>
<tr valign="top"><td><code>margins</code></td>
<td>
<p>numeric vector of length two containing the margins (see <code><a href="../../graphics/html/par.html">par</a>(mar)</code>)
for column and row names respectively.</p>
</td></tr>
<tr valign="top"><td><code>lhei, lwid</code></td>
<td>
<p>arguments passed to <code>layout</code> to divide the device up into two
(or three if a side color is drawn) rows and two columns, with the row-heights <code>lhei</code>
and the column-widths <code>lwid</code>.</p>
</td></tr>
<tr valign="top"><td><code>legend</code></td>
<td>
<p>A list indicating the legend for each group, the color vector,  title of the legend and cex.</p>
</td></tr>
<tr valign="top"><td><code>save</code></td>
<td>
<p>should the plot be saved? If so, argument to be set to either  <code>'jpeg'</code>, <code>'tiff'</code>, <code>'png'</code> or <code>'pdf'</code>.</p>
</td></tr>
<tr valign="top"><td><code>name.save</code></td>
<td>
<p>character string for the name of the file to be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>One matrix Clustered Image Map (default method) is a 2-dimensional visualization
of a real-valued matrix (basically <code><a href="../../graphics/html/image.html">image</a>(t(mat))</code>) with rows and/or
columns reordered according to some hierarchical clustering method to identify
interesting patterns. Generated dendrograms from clustering are added to the left
side and to the top of the image. By default the used clustering method for rows
and columns is the <em>complete linkage</em> method and the used distance measure
is the distance <em>euclidean</em>.
</p>
<p>In <code>"pca"</code>, <code>"spca"</code>, <code>"ipca"</code>, <code>"sipca"</code>, <code>"plsda"</code>,
<code>"splsda"</code> and multilevel variants methods the <code>mat</code> matrix is <code>object$X</code>.
</p>
<p>For the remaining methods, if <code>mapping = "X"</code> or <code>mapping = "Y"</code> the
<code>mat</code> matrix is <code>object$X</code> or <code>object$Y</code> respectively. If <code>mapping = "XY"</code>:
</p>

<ul>
<li><p> in <code>rcc</code> method, the matrix <code>mat</code> is created where element <i>(j,k)</i>
is the scalar product value between every pairs of vectors in dimension
<code>length(comp)</code> representing the variables <i>X_j</i> and <i>Y_k</i> on the
axis defined by <i>Z_i</i> with <i>i</i> in <code>comp</code>, where <i>Z_i</i> is the
equiangular vector between the <i>i</i>-th <i>X</i> and <i>Y</i> canonical variate.
</p>
</li>
<li><p> in <code>pls</code>, <code>spls</code> and multilevel spls methods, if <code>object$mode</code> is
<code>"regression"</code>, the element <i>(j,k)</i> of the matrix <code>mat</code>
is given by the scalar product value between every pairs of vectors in dimension
<code>length(comp)</code> representing the variables <i>X_j</i> and <i>Y_k</i> on the axis
defined by <i>U_i</i> with <i>i</i> in <code>comp</code>, where <i>U_i</i> is the <i>i</i>-th
<i>X</i> variate. If <code>object$mode</code> is <code>"canonical"</code> then <i>X_j</i> and
<i>Y_k</i> are represented on the axis defined by <i>U_i</i> and <i>V_i</i> respectively.</p>
</li></ul>

<p>By default four components will be displayed in the plot. At the top left is the
color key, top right is the column dendogram, bottom left is the row dendogram,
bottom right is the image plot. When <code>sideColors</code> are provided, an
additional row or column is inserted in the appropriate location. This layout can
be overriden by specifiying appropriate values for <code>lwid</code> and <code>lhei</code>.
<code>lwid</code> controls the column width, and <code>lhei</code> controls the row height.
See the help page for <code><a href="../../graphics/html/layout.html">layout</a></code> for details on how to use these arguments.
</p>
<p>For visualization of &quot;high-dimensional&quot; data sets, a nice zooming tool was created.
<code>zoom = TRUE</code> open a new device, one for CIM, one for zoom-out region and
define an interactive 'zoom' process: click two points at imagen map region by
pressing the first mouse button. It then draws a rectangle around the selected
region and zoom-out this at new device. The process can be repeated to zoom-out
other regions of interest.
</p>
<p>The zoom process is terminated by clicking the second button and selecting 'Stop'
from the menu, or from the 'Stop' menu on the graphics window.
</p>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>M</code></td>
<td>
<p>the mapped matrix used by <code>cim</code>.</p>
</td></tr>
<tr valign="top"><td><code>rowInd, colInd</code></td>
<td>
<p>row and column index permutation vectors as returned
by <code><a href="../../stats/html/order.dendrogram.html">order.dendrogram</a></code>.</p>
</td></tr>
<tr valign="top"><td><code>ddr, ddc</code></td>
<td>
<p>object of class <code>"dendrogram"</code> which describes the row and
column trees produced by <code>cim</code>.</p>
</td></tr>
<tr valign="top"><td><code>mat.cor</code></td>
<td>
<p>the correlation matrix used for the heatmap. Available only when mapping = &quot;XY&quot;.</p>
</td></tr>
<tr valign="top"><td><code>row.names, col.names</code></td>
<td>
<p>character vectors with row and column labels used.</p>
</td></tr>
<tr valign="top"><td><code>row.sideColors, col.sideColors</code></td>
<td>
<p>character vector containing the color
names for vertical and horizontal side bars used to annotate the rows and columns.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ignacio González, Francois Bartolo, Kim-Anh Lê Cao.</p>


<h3>References</h3>

<p>Eisen, M. B., Spellman, P. T., Brown, P. O. and Botstein, D. (1998).
Cluster analysis and display of genome-wide expression patterns.
<em>Proceeding of the National Academy of Sciences of the USA</em> <b>95</b>, 14863-14868.
</p>
<p>Weinstein, J. N., Myers, T. G., O'Connor, P. M., Friend, S. H., Fornace Jr., A. J.,
Kohn, K. W., Fojo, T., Bates, S. E., Rubinstein, L. V., Anderson, N. L.,
Buolamwini, J. K., van Osdol, W. W., Monks, A. P., Scudiero, D. A., Sausville, E. A.,
Zaharevitz, D. W., Bunow, B., Viswanadhan, V. N., Johnson, G. S., Wittes, R. E.
and Paull, K. D. (1997). An information-intensive approach to the molecular
pharmacology of cancer. <em>Science</em> <b>275</b>, 343-349.
</p>
<p>González I., Lê Cao K.A., Davis M.J., Déjean S. (2012). Visualising associations between
paired 'omics' data sets. <em>BioData Mining</em>; <b>5</b>(1).
</p>
<p>mixOmics article:
</p>
<p>Rohart F, Gautier B, Singh A, Lê Cao K-A. mixOmics: an R package for 'omics feature selection and multiple data integration. PLoS Comput Biol 13(11): e1005752
</p>


<h3>See Also</h3>

<p><code><a href="../../stats/html/heatmap.html">heatmap</a></code>,
<code><a href="../../stats/html/hclust.html">hclust</a></code>, <code><a href="plotVar.html">plotVar</a></code>,
<code><a href="network.html">network</a></code> and
</p>
<p><a href="http://mixomics.org/graphics/">http://mixomics.org/graphics/</a> for more details on all options available.</p>


<h3>Examples</h3>

<pre>
## default method: shows cross correlation between 2 data sets
#------------------------------------------------------------------
data(nutrimouse)
X &lt;- nutrimouse$lipid
Y &lt;- nutrimouse$gene

cim(cor(X, Y), cluster = "none")


## Not run: 
## CIM representation for objects of class 'rcc'
#------------------------------------------------------------------

nutri.rcc &lt;- rcc(X, Y, ncomp = 3, lambda1 = 0.064, lambda2 = 0.008)

cim(nutri.rcc, xlab = "genes", ylab = "lipids", margins = c(5, 6))

#-- interactive 'zoom' available as below

cim(nutri.rcc, xlab = "genes", ylab = "lipids", margins = c(5, 6),
zoom = TRUE)
#-- select the region and "see" the zoom-out region


#-- cim from X matrix with a side bar to indicate the diet
diet.col &lt;- palette()[as.numeric(nutrimouse$diet)]
cim(nutri.rcc, mapping = "X", row.names = nutrimouse$diet,
row.sideColors = diet.col, xlab = "lipids",
clust.method = c("ward", "ward"), margins = c(6, 4))

#-- cim from Y matrix with a side bar to indicate the genotype
geno.col = color.mixo(as.numeric(nutrimouse$genotype))
cim(nutri.rcc, mapping = "Y", row.names = nutrimouse$genotype,
row.sideColors = geno.col, xlab = "genes",
clust.method = c("ward", "ward"))

#-- save the result as a jpeg file
jpeg(filename = "test.jpeg", res = 600, width = 4000, height = 4000)
cim(nutri.rcc, xlab = "genes", ylab = "lipids", margins = c(5, 6))
dev.off()

## CIM representation for objects of class 'spca' (also works for sipca)
#------------------------------------------------------------------

data(liver.toxicity)
X &lt;- liver.toxicity$gene

liver.spca &lt;- spca(X, ncomp = 2, keepX = c(30, 30), scale = FALSE)

dose.col &lt;- color.mixo(as.numeric(as.factor(liver.toxicity$treatment[, 3])))

# side bar, no variable names shown
cim(liver.spca, row.sideColors = dose.col, col.names = FALSE,
row.names = liver.toxicity$treatment[, 3],
clust.method = c("ward", "ward"))


## CIM representation for objects of class '(s)pls'
#------------------------------------------------------------------

data(liver.toxicity)

X &lt;- liver.toxicity$gene
Y &lt;- liver.toxicity$clinic
liver.spls &lt;- spls(X, Y, ncomp = 3,
keepX = c(20, 50, 50), keepY = c(10, 10, 10))


# default
cim(liver.spls)


# transpose matrix, choose clustering method
cim(liver.spls, transpose = TRUE,
clust.method = c("ward", "ward"), margins = c(5, 7))

# Here we visualise only the X variables selected
cim(liver.spls, mapping="X")

# Here we should visualise only the Y variables selected
cim(liver.spls, mapping="Y")

# Here we only visualise the similarity matrix between the variables by spls
cim(liver.spls, cluster="none")

# plotting two data sets with the similarity matrix as input in the funciton
# (see our BioData Mining paper for more details)
# Only the variables selected by the sPLS model in X and Y are represented
cim(liver.spls, mapping="XY")

# on the X matrix only, side col var to indicate dose
dose.col &lt;- color.mixo(as.numeric(as.factor(liver.toxicity$treatment[, 3])))
cim(liver.spls, mapping = "X", row.sideColors = dose.col,
row.names = liver.toxicity$treatment[, 3])

# CIM default representation includes the total of 120 genes selected, with the dose color
# with a sparse method, show only the variables selected on specific components
cim(liver.spls, comp = 1)
cim(liver.spls, comp = 2)
cim(liver.spls, comp = c(1,2))
cim(liver.spls, comp = c(1,3))


## CIM representation for objects of class '(s)plsda'
#------------------------------------------------------------------
data(liver.toxicity)

X &lt;- liver.toxicity$gene
# Setting up the Y outcome first
Y &lt;- liver.toxicity$treatment[, 3]
#set up colors for cim
dose.col &lt;- color.mixo(as.numeric(as.factor(liver.toxicity$treatment[, 3])))


liver.splsda &lt;- splsda(X, Y, ncomp = 2, keepX = c(40, 30))

cim(liver.splsda, row.sideColors = dose.col, row.names = Y)


## CIM representation for objects of class splsda 'multilevel'
# with a two level factor (repeated sample and time)
#------------------------------------------------------------------
data(vac18.simulated)
X &lt;- vac18.simulated$genes
design &lt;- data.frame(samp = vac18.simulated$sample)
Y = data.frame(time = vac18.simulated$time,
stim = vac18.simulated$stimulation)

res.2level &lt;- splsda(X, Y = Y, ncomp = 2, multilevel = design,
keepX = c(120, 10))

#define colors for the levels: stimulation and time
stim.col &lt;- c("darkblue", "purple", "green4","red3")
stim.col &lt;- stim.col[as.numeric(Y$stim)]
time.col &lt;- c("orange", "cyan")[as.numeric(Y$time)]


# The row side bar indicates the two levels of the facteor, stimulation and time.
# the sample names have been motified on the plot.
cim(res.2level, row.sideColors = cbind(stim.col, time.col),
row.names = paste(Y$time, Y$stim, sep = "_"),
col.names = FALSE,
#setting up legend:
legend=list(legend = c(levels(Y$time), levels(Y$stim)),
col = c("orange", "cyan", "darkblue", "purple", "green4","red3"),
title = "Condition", cex = 0.7)
)


## CIM representation for objects of class spls 'multilevel'
#------------------------------------------------------------------

data(liver.toxicity)
repeat.indiv &lt;- c(1, 2, 1, 2, 1, 2, 1, 2, 3, 3, 4, 3, 4, 3, 4, 4, 5, 6, 5, 5,
6, 5, 6, 7, 7, 8, 6, 7, 8, 7, 8, 8, 9, 10, 9, 10, 11, 9, 9,
10, 11, 12, 12, 10, 11, 12, 11, 12, 13, 14, 13, 14, 13, 14,
13, 14, 15, 16, 15, 16, 15, 16, 15, 16)

# sPLS is a non supervised technique, and so we only indicate the sample repetitions
# in the design (1 factor only here, sample)
# sPLS takes as an input 2 data sets, and the variables selected
design &lt;- data.frame(sample = repeat.indiv)
res.spls.1level &lt;- spls(X = liver.toxicity$gene,
Y=liver.toxicity$clinic,
multilevel = design,
ncomp = 2,
keepX = c(50, 50), keepY = c(5, 5),
mode = 'canonical')

stim.col &lt;- c("darkblue", "purple", "green4","red3")

# showing only the Y variables, and only those selected in comp 1
cim(res.spls.1level, mapping="Y",
row.sideColors = stim.col[factor(liver.toxicity$treatment[,3])], comp = 1,
#setting up legend:
legend=list(legend = unique(liver.toxicity$treatment[,3]), col=stim.col,
title = "Dose", cex=0.9))


# showing only the X variables, for all selected on comp 1 and 2
cim(res.spls.1level, mapping="X",
row.sideColors = stim.col[factor(liver.toxicity$treatment[,3])],
#setting up legend:
legend=list(legend = unique(liver.toxicity$treatment[,3]), col=stim.col,
title = "Dose", cex=0.9))


# These are the cross correlations between the variables selected in X and Y.
# The similarity matrix is obtained as in our paper in Data Mining
cim(res.spls.1level, mapping="XY")


## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.6.2 <a href="00Index.html">Index</a>]</div>
</body></html>
