<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: 16S microbiome data: most diverse bodysites from HMP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for diverse.16S {mixOmics}"><tr><td>diverse.16S {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>16S microbiome data: most diverse bodysites from HMP</h2>

<h3>Description</h3>

<p>The 16S data from the Human Microbiome Project includes only the most
diverse bodysites: Antecubital fossa (skin), Stool and Subgingival plaque
(oral) and can be analysed using a multilevel approach to account for
repeated measurements using our module mixMC. The data include 162 samples
(54 unique healthy individuals) measured on 1,674 OTUs.
</p>


<h3>Usage</h3>

<pre>
data(diverse.16S)
</pre>


<h3>Format</h3>

<p>A list containing two data sets, <code>data.TSS</code> and <code>data.raw</code>
and some meta data information: </p>
 <dl>
<dt>list(&quot;data.TSS&quot;)</dt><dd><p>data
frame with 162 rows (samples) and 1674 columns (OTUs). The prefiltered
normalised data using Total Sum Scaling normalisation.</p>
</dd>
<dt>list(&quot;data.raw&quot;)</dt><dd><p>data frame with 162 rows (samples) and 1674 columns
(OTUs). The prefiltered raw count OTU data which include a 1 offset (i.e. no
0 values).</p>
</dd> <dt>list(&quot;taxonomy&quot;)</dt><dd><p>data frame with 1674 rows (OTUs) and 6
columns indicating the taxonomy of each OTU.</p>
</dd> <dt>list(&quot;indiv&quot;)</dt><dd><p>data
frame with 162 rows indicating sample meta data.</p>
</dd>
<dt>list(&quot;bodysite&quot;)</dt><dd><p>factor of length 162 indicating the bodysite with
levels &quot;Antecubital_fossa&quot;, &quot;Stool&quot; and &quot;Subgingival_plaque&quot;.</p>
</dd>
<dt>list(&quot;sample&quot;)</dt><dd><p>vector of length 162 indicating the unique individual
ID, useful for a multilevel approach to taken into account the repeated
measured on each individual.</p>
</dd> </dl>



<h3>Details</h3>

<p>The data were downloaded from the Human Microbiome Project (HMP,
http://hmpdacc.org/HMQCP/all/ for the V1-3 variable region).  The original
data contained 43,146 OTU counts for 2,911 samples measured from 18
different body sites. We focused on the first visit of each healthy
individual and focused on the three most diverse habitats. The prefiltered
dataset included 1,674 OTU counts. We strongly recommend to use log ratio
transformations on the <code>data.TSS</code> normalised data, as implemented in
the PLS and PCA methods, see details on <a href="www.mixOmics.org/mixMC">www.mixOmics.org/mixMC</a>.
</p>
<p>The <code>data.raw</code> include a 1 offset in order to be log ratios transformed
after TSS normalisation. Consequently, the <code>data.TSS</code> are TSS
normalisation of <code>data.raw</code>. The CSS normalisation was performed on the
orignal data (including zero values)
</p>


<h3>Value</h3>

<p>none
</p>


<h3>Source</h3>

<p>The raw data were downloaded from
<a href="http://hmpdacc.org/HMQCP/all/">http://hmpdacc.org/HMQCP/all/</a>. Filtering and normalisation described
in our website <a href="www.mixOmics.org/mixMC">www.mixOmics.org/mixMC</a>
</p>


<h3>References</h3>

<p>LÃª Cao K.-A., Costello ME, Lakis VA, Bartolo, F,Chua XY,
Brazeilles R, Rondeau P. MixMC: Multivariate insights into Microbial
Communities. PLoS ONE, 11(8): e0160169 (2016).
</p>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.18.1 <a href="00Index.html">Index</a>]</div>
</div></body></html>
