<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: P-integration with Discriminant Analysis and variable...</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for mint.splsda {mixOmics}"><tr><td>mint.splsda {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>P-integration with Discriminant Analysis and variable selection</h2>

<h3>Description</h3>

<p>Function to combine multiple independent studies measured on the same
variables or predictors (P-integration) using variants of multi-group sparse
PLS-DA for supervised classification with variable selection.
</p>


<h3>Usage</h3>

<pre>
mint.splsda(
  X,
  Y,
  ncomp = 2,
  study,
  keepX = rep(ncol(X), ncomp),
  scale = TRUE,
  tol = 1e-06,
  max.iter = 100,
  near.zero.var = FALSE,
  all.outputs = TRUE
)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>numeric matrix of predictors combining multiple independent studies
on the same set of predictors. <code>NA</code>s are allowed.</p>
</td></tr>
<tr valign="top"><td><code>Y</code></td>
<td>
<p>A factor or a class vector indicating the discrete outcome of each
sample.</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>Positive Integer. The number of components to include in the
model. Default to 2.</p>
</td></tr>
<tr valign="top"><td><code>study</code></td>
<td>
<p>Factor, indicating the membership of each sample to each of the
studies being combined</p>
</td></tr>
<tr valign="top"><td><code>keepX</code></td>
<td>
<p>numeric vector indicating the number of variables to select in
<code>X</code> on each component. By default all variables are kept in the model.</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>Logical. If scale = TRUE, each block is standardized to zero
means and unit variances (default: TRUE)</p>
</td></tr>
<tr valign="top"><td><code>tol</code></td>
<td>
<p>Positive numeric used as convergence criteria/tolerance during the
iterative process. Default to <code>1e-06</code>.</p>
</td></tr>
<tr valign="top"><td><code>max.iter</code></td>
<td>
<p>Integer, the maximum number of iterations. Default to  100.</p>
</td></tr>
<tr valign="top"><td><code>near.zero.var</code></td>
<td>
<p>Logical, see the internal <code><a href="../../mixOmics/help/nearZeroVar.html">nearZeroVar</a></code>
function (should be set to TRUE in particular for data with many zero
values). Setting this argument to FALSE (when appropriate) will speed up the
computations. Default value is FALSE.</p>
</td></tr>
<tr valign="top"><td><code>all.outputs</code></td>
<td>
<p>Logical. Computation can be faster when some specific
(and non-essential) outputs are not calculated. Default = <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mint.splsda</code> function fits a vertical sparse PLS-DA models with
<code>ncomp</code> components in which several independent studies measured on the
same variables are integrated. The aim is to classify the discrete outcome
<code>Y</code> and select variables that explain the outcome. The <code>study</code>
factor indicates the membership of each sample in each study. We advise to
only combine studies with more than 3 samples as the function performs
internal scaling per study, and where all outcome categories are
represented.
</p>
<p><code>X</code> can contain missing values. Missing values are handled by being
disregarded during the cross product computations in the algorithm
<code>mint.splsda</code> without having to delete rows with missing data.
Alternatively, missing data can be imputed prior using the
<code><a href="../../mixOmics/help/impute.nipals.html">impute.nipals</a></code> function.
</p>
<p>The type of deflation used is <code>'regression'</code> for discriminant algorithms.
i.e. no deflation is performed on Y.
</p>
<p>Variable selection is performed on each component for <code>X</code> via input
parameter <code>keepX</code>.
</p>
<p>Useful graphical outputs are available, e.g. <code><a href="../../mixOmics/help/plotIndiv.html">plotIndiv</a></code>,
<code><a href="../../mixOmics/help/plotLoadings.html">plotLoadings</a></code>, <code><a href="../../mixOmics/help/plotVar.html">plotVar</a></code>.
</p>


<h3>Value</h3>

<p><code>mint.splsda</code> returns an object of class <code>"mint.splsda",
"splsda"</code>, a list that contains the following components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>the centered and standardized original predictor matrix.</p>
</td></tr>
<tr valign="top"><td><code>Y</code></td>
<td>
<p>the centered and standardized original response vector or matrix.</p>
</td></tr>
<tr valign="top"><td><code>ind.mat</code></td>
<td>
<p>the centered and standardized original response vector or
matrix.</p>
</td></tr> <tr valign="top"><td><code>ncomp</code></td>
<td>
<p>the number of components included in the model.</p>
</td></tr>
<tr valign="top"><td><code>study</code></td>
<td>
<p>The study grouping factor</p>
</td></tr> <tr valign="top"><td><code>mode</code></td>
<td>
<p>the algorithm used to
fit the model.</p>
</td></tr> <tr valign="top"><td><code>keepX</code></td>
<td>
<p>Number of variables used to build each
component of X</p>
</td></tr> <tr valign="top"><td><code>variates</code></td>
<td>
<p>list containing the variates of X - global
variates.</p>
</td></tr> <tr valign="top"><td><code>loadings</code></td>
<td>
<p>list containing the estimated loadings for the
variates - global loadings.</p>
</td></tr> <tr valign="top"><td><code>variates.partial</code></td>
<td>
<p>list containing the
variates of X relative to each study - partial variates.</p>
</td></tr>
<tr valign="top"><td><code>loadings.partial</code></td>
<td>
<p>list containing the estimated loadings for the
partial variates - partial loadings.</p>
</td></tr> <tr valign="top"><td><code>names</code></td>
<td>
<p>list containing the names
to be used for individuals and variables.</p>
</td></tr> <tr valign="top"><td><code>nzv</code></td>
<td>
<p>list containing the
zero- or near-zero predictors information.</p>
</td></tr> <tr valign="top"><td><code>iter</code></td>
<td>
<p>Number of iterations
of the algorithm for each component</p>
</td></tr> <tr valign="top"><td><code>prop_expl_var</code></td>
<td>
<p>Percentage of
explained variance for each component and each study (note that contrary to
PCA, this amount may not decrease as the aim of the method is not to
maximise the variance, but the covariance between X and the dummy matrix
Y).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Florian Rohart, Kim-Anh Lê Cao, Al J Abadi
</p>


<h3>References</h3>

<p>Rohart F, Eslami A, Matigian, N, Bougeard S, Lê Cao K-A (2017).
MINT: A multivariate integrative approach to identify a reproducible
biomarker signature across multiple experiments and platforms. BMC
Bioinformatics 18:128.
</p>
<p>Eslami, A., Qannari, E. M., Kohler, A., and Bougeard, S. (2014). Algorithms
for multi-group PLS. J. Chemometrics, 28(3), 192-201.
</p>
<p>mixOmics article:
</p>
<p>Rohart F, Gautier B, Singh A, Lê Cao K-A. mixOmics: an R package for 'omics
feature selection and multiple data integration. PLoS Comput Biol 13(11):
e1005752
</p>


<h3>See Also</h3>

<p><code><a href="../../mixOmics/help/spls.html">spls</a></code>, <code><a href="../../mixOmics/help/summary.html">summary</a></code>, <code><a href="../../mixOmics/help/plotIndiv.html">plotIndiv</a></code>,
<code><a href="../../mixOmics/help/plotVar.html">plotVar</a></code>, <code><a href="../../mixOmics/help/predict.html">predict</a></code>, <code><a href="../../mixOmics/help/perf.html">perf</a></code>,
<code><a href="../../mixOmics/help/mint.pls.html">mint.pls</a></code>, <code><a href="../../mixOmics/help/mint.plsda.html">mint.plsda</a></code>, <code><a href="../../mixOmics/help/mint.plsda.html">mint.plsda</a></code>
and http://www.mixOmics.org/mixMINT for more details.
</p>


<h3>Examples</h3>

<pre>

data(stemcells)

# -- feature selection
res = mint.splsda(X = stemcells$gene, Y = stemcells$celltype, ncomp = 3, keepX = c(10, 5, 15),
study = stemcells$study)

plotIndiv(res)
#plot study-specific outputs for all studies
plotIndiv(res, study = "all.partial")

## Not run: 
#plot study-specific outputs for study "2"
plotIndiv(res, study = "2")

#plot study-specific outputs for study "2", "3" and "4"
plotIndiv(res, study = c(2, 3, 4))

## End(Not run)
</pre>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.18.1 <a href="00Index.html">Index</a>]</div>
</div></body></html>
