<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: NP-integration with Discriminant Analysis</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for mint.block.plsda {mixOmics}"><tr><td>mint.block.plsda {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>NP-integration with Discriminant Analysis</h2>

<h3>Description</h3>

<p>Function to integrate data sets measured on the same samples (N-integration)
and to combine multiple independent studies measured on the same variables
or predictors (P-integration) using variants of multi-group and generalised
PLS-DA for supervised classification.
</p>


<h3>Usage</h3>

<pre>
mint.block.plsda(
  X,
  Y,
  indY,
  study,
  ncomp = 2,
  design,
  scheme,
  scale = TRUE,
  init,
  tol = 1e-06,
  max.iter = 100,
  near.zero.var = FALSE,
  all.outputs = TRUE
)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>A named list of data sets (called 'blocks') measured on the same samples.
Data in the list should be arranged in samples x variables, with samples
order matching in all data sets.</p>
</td></tr>
<tr valign="top"><td><code>Y</code></td>
<td>
<p>A factor or a class vector indicating the discrete outcome of each
sample.</p>
</td></tr>
<tr valign="top"><td><code>indY</code></td>
<td>
<p>To be supplied if Y is missing, indicates the position of the
matrix / vector response in the list <code>X</code></p>
</td></tr>
<tr valign="top"><td><code>study</code></td>
<td>
<p>Factor, indicating the membership of each sample to each of the
studies being combined</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>the number of components to include in the model. Default to 2.
Applies to all blocks.</p>
</td></tr>
<tr valign="top"><td><code>design</code></td>
<td>
<p>numeric matrix of size (number of blocks in X) x (number of
blocks in X) with values between 0 and 1. Each value indicates the strenght
of the relationship to be modelled between two blocks; a value of 0
indicates no relationship, 1 is the maximum value. Alternatively, one of
c('null', 'full') indicating a disconnected or fully connected design,
respecively, or a numeric between 0 and 1 which will designate all
off-diagonal elements of a fully connected design (see examples in
<code>block.splsda</code>). If <code>Y</code> is provided instead of <code>indY</code>, the
<code>design</code> matrix is changed to include relationships to <code>Y</code>.</p>
</td></tr>
<tr valign="top"><td><code>scheme</code></td>
<td>
<p>Character, one of 'horst', 'factorial' or 'centroid'. Default =
<code>'horst'</code>, see reference.</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>Logical. If scale = TRUE, each block is standardized to zero
means and unit variances (default: TRUE)</p>
</td></tr>
<tr valign="top"><td><code>init</code></td>
<td>
<p>Mode of initialization use in the algorithm, either by Singular
Value Decomposition of the product of each block of X with Y ('svd') or each
block independently ('svd.single'). Default = <code>svd.single</code>.</p>
</td></tr>
<tr valign="top"><td><code>tol</code></td>
<td>
<p>Positive numeric used as convergence criteria/tolerance during the
iterative process. Default to <code>1e-06</code>.</p>
</td></tr>
<tr valign="top"><td><code>max.iter</code></td>
<td>
<p>Integer, the maximum number of iterations. Default to  100.</p>
</td></tr>
<tr valign="top"><td><code>near.zero.var</code></td>
<td>
<p>Logical, see the internal <code><a href="../../mixOmics/help/nearZeroVar.html">nearZeroVar</a></code>
function (should be set to TRUE in particular for data with many zero
values). Setting this argument to FALSE (when appropriate) will speed up the
computations. Default value is FALSE.</p>
</td></tr>
<tr valign="top"><td><code>all.outputs</code></td>
<td>
<p>Logical. Computation can be faster when some specific
(and non-essential) outputs are not calculated. Default = <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function fits multi-group generalised PLS models with a specified number
of <code>ncomp</code> components. A factor indicating the discrete outcome needs
to be provided, either by <code>Y</code> or by its position <code>indY</code> in the
list of blocks <code>X</code>.
</p>
<p><code>X</code> can contain missing values. Missing values are handled by being
disregarded during the cross product computations in the algorithm
<code>block.pls</code> without having to delete rows with missing data.
Alternatively, missing data can be imputed prior using the
<code><a href="../../mixOmics/help/impute.nipals.html">impute.nipals</a></code> function.
</p>
<p>The type of algorithm to use is specified with the <code>mode</code> argument.
Four PLS algorithms are available: PLS regression <code>("regression")</code>, PLS
canonical analysis <code>("canonical")</code>, redundancy analysis
<code>("invariant")</code> and the classical PLS algorithm <code>("classic")</code> (see
References and more details in <code>?pls</code>).
</p>


<h3>Value</h3>

<p><code>mint.block.plsda</code> returns an object of class
<code>"mint.plsda", "block.plsda"</code>, a list that contains the following
components:
</p>
<table summary="R valueblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>the centered and standardized original predictor matrix.</p>
</td></tr>
<tr valign="top"><td><code>Y</code></td>
<td>
<p>the centered and standardized original response vector or matrix.</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>the number of components included in the model for each block.</p>
</td></tr>
<tr valign="top"><td><code>mode</code></td>
<td>
<p>the algorithm used to fit the model.</p>
</td></tr> <tr valign="top"><td><code>mat.c</code></td>
<td>
<p>matrix of
coefficients from the regression of X / residual matrices X on the
X-variates, to be used internally by <code>predict</code>.</p>
</td></tr> <tr valign="top"><td><code>variates</code></td>
<td>
<p>list
containing the <i>X</i> and <i>Y</i> variates.</p>
</td></tr> <tr valign="top"><td><code>loadings</code></td>
<td>
<p>list
containing the estimated loadings for the variates.</p>
</td></tr> <tr valign="top"><td><code>names</code></td>
<td>
<p>list
containing the names to be used for individuals and variables.</p>
</td></tr>
<tr valign="top"><td><code>nzv</code></td>
<td>
<p>list containing the zero- or near-zero predictors information.</p>
</td></tr>
<tr valign="top"><td><code>tol</code></td>
<td>
<p>the tolerance used in the iterative algorithm, used for
subsequent S3 methods</p>
</td></tr> <tr valign="top"><td><code>max.iter</code></td>
<td>
<p>the maximum number of iterations,
used for subsequent S3 methods</p>
</td></tr> <tr valign="top"><td><code>iter</code></td>
<td>
<p>Number of iterations of the
algorithm for each component</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Florian Rohart, Benoit Gautier, Kim-Anh Lê Cao, Al J Abadi
</p>


<h3>References</h3>

<p>On multi-group PLS:
</p>
<p>Rohart F, Eslami A, Matigian, N, Bougeard S, Lê Cao K-A (2017). MINT: A
multivariate integrative approach to identify a reproducible biomarker
signature across multiple experiments and platforms. BMC Bioinformatics
18:128.
</p>
<p>Eslami, A., Qannari, E. M., Kohler, A., and Bougeard, S. (2014). Algorithms
for multi-group PLS. J. Chemometrics, 28(3), 192-201.
</p>
<p>On multiple integration with PLSDA:
</p>
<p>Singh A., Gautier B., Shannon C., Vacher M., Rohart F., Tebbutt S. and Lê
Cao K.A. (2016). DIABLO: multi omics integration for biomarker discovery.
BioRxiv available here:
<a href="http://biorxiv.org/content/early/2016/08/03/067611">http://biorxiv.org/content/early/2016/08/03/067611</a> Tenenhaus A.,
Philippe C., Guillemot V, Lê Cao K.A., Grill J, Frouin V. Variable selection
for generalized canonical correlation analysis. <em>Biostatistics</em>. kxu001
</p>
<p>Gunther O., Shin H., Ng R. T. , McMaster W. R., McManus B. M. , Keown P. A.
, Tebbutt S.J. , Lê Cao K-A. , (2014) Novel multivariate methods for
integration of genomics and proteomics data: Applications in a kidney
transplant rejection study, OMICS: A journal of integrative biology, 18(11),
682-95.
</p>
<p>mixOmics article:
</p>
<p>Rohart F, Gautier B, Singh A, Lê Cao K-A. mixOmics: an R package for 'omics
feature selection and multiple data integration. PLoS Comput Biol 13(11):
e1005752
</p>


<h3>See Also</h3>

<p><code><a href="../../mixOmics/help/spls.html">spls</a></code>, <code><a href="../../mixOmics/help/summary.html">summary</a></code>, <code><a href="../../mixOmics/help/plotIndiv.html">plotIndiv</a></code>,
<code><a href="../../mixOmics/help/plotVar.html">plotVar</a></code>, <code><a href="../../mixOmics/help/predict.html">predict</a></code>, <code><a href="../../mixOmics/help/perf.html">perf</a></code>,
<code><a href="../../mixOmics/help/mint.block.spls.html">mint.block.spls</a></code>, <code><a href="../../mixOmics/help/mint.block.plsda.html">mint.block.plsda</a></code>,
<code><a href="../../mixOmics/help/mint.block.splsda.html">mint.block.splsda</a></code> and http://www.mixOmics.org/mixMINT for more
details.
</p>


<h3>Examples</h3>

<pre>

data(breast.TCGA)

# for the purpose of this example, we consider the training set as study1 and
# the test set as another independent study2.
study = c(rep("study1",150), rep("study2",70))

mrna = rbind(breast.TCGA$data.train$mrna, breast.TCGA$data.test$mrna)
mirna = rbind(breast.TCGA$data.train$mirna, breast.TCGA$data.test$mirna)
data = list(mrna = mrna, mirna = mirna)

Y = c(breast.TCGA$data.train$subtype, breast.TCGA$data.test$subtype)

res = mint.block.plsda(data,Y,study=study, ncomp=2)

res
</pre>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.18.1 <a href="00Index.html">Index</a>]</div>
</div></body></html>
