<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: 'Omics Data Integration Project</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for mixOmics-package {mixOmics}"><tr><td>mixOmics-package {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>'Omics Data Integration Project</h2>

<h3>Description</h3>

<p>Multivariate methods are well suited to large omics data sets where the
number of variables (e.g. genes, proteins, metabolites) is much larger than
the number of samples (patients, cells, mice). They have the appealing
properties of reducing the dimension of the data by using instrumental
variables (components), which are defined as combinations of all variables.
Those components are then used to produce useful graphical outputs that
enable better understanding of the relationships and correlation structures
between the different data sets that are integrated.
</p>


<h3>Details</h3>

<p>mixOmics offers a wide
range of multivariate methods for the exploration and integration of
biological datasets with a particular focus on variable selection. The
package proposes several sparse multivariate models we have developed to
identify the key variables that are highly correlated, and/or explain the
biological outcome of interest. The data that can be analysed with mixOmics
may come from high throughput sequencing technologies, such as omics data
(transcriptomics, metabolomics, proteomics, metagenomics etc) but also beyond
the realm of omics (e.g. spectral imaging). 
</p>
<p>The methods implemented in
mixOmics can also handle missing values without having to delete entire rows
with missing data. A non exhaustive list of methods include variants of
generalised Canonical Correlation Analysis, sparse Partial Least Squares and
sparse Discriminant Analysis. Recently we implemented integrative methods to
combine multiple data sets: N-integration with variants of Generalised
Canonical Correlation Analysis and P-integration with variants of multi-group
Partial Least Squares.
</p>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.18.1 <a href="00Index.html">Index</a>]</div>
</div></body></html>
