<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head><title>R: Sparse Principal Components Analysis</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link rel="stylesheet" type="text/css" href="R.css" />
</head><body><div class="container">

<table width="100%" summary="page for spca {mixOmics}"><tr><td>spca {mixOmics}</td><td style="text-align: right;">R Documentation</td></tr></table>

<h2>Sparse Principal Components Analysis</h2>

<h3>Description</h3>

<p>Performs a sparse principal component analysis for variable selection using
singular value decomposition and lasso penalisation on the loading vectors.
</p>


<h3>Usage</h3>

<pre>
spca(
  X,
  ncomp = 2,
  center = TRUE,
  scale = TRUE,
  keepX = rep(ncol(X), ncomp),
  max.iter = 500,
  tol = 1e-06,
  logratio = c("none", "CLR"),
  multilevel = NULL
)
</pre>


<h3>Arguments</h3>

<table summary="R argblock">
<tr valign="top"><td><code>X</code></td>
<td>
<p>a numeric matrix (or data frame) which provides the data for the sparse
principal components analysis. It should not contain missing values.</p>
</td></tr>
<tr valign="top"><td><code>ncomp</code></td>
<td>
<p>Integer, if data is complete <code>ncomp</code> decides the number of
components and associated eigenvalues to display from the <code>pcasvd</code>
algorithm and if the data has missing values, <code>ncomp</code> gives the number
of components to keep to perform the reconstitution of the data using the
NIPALS algorithm. If <code>NULL</code>, function sets <code>ncomp = min(nrow(X),
ncol(X))</code></p>
</td></tr>
<tr valign="top"><td><code>center</code></td>
<td>
<p>(Default=TRUE) Logical, whether the variables should be shifted
to be zero centered. Only set to FALSE if data have already been centered.
Alternatively, a vector of length equal the number of columns of <code>X</code>
can be supplied. The value is passed to <code><a href="../../base/html/scale.html">scale</a></code>. If the data
contain missing values, columns should be centered for reliable results.</p>
</td></tr>
<tr valign="top"><td><code>scale</code></td>
<td>
<p>(Default=TRUE) Logical indicating whether the variables should be
scaled to have unit variance before the analysis takes place.</p>
</td></tr>
<tr valign="top"><td><code>keepX</code></td>
<td>
<p>numeric vector of length <code>ncomp</code>, the number of variables to keep
in loading vectors. By default all variables are kept in the model. See
details.</p>
</td></tr>
<tr valign="top"><td><code>max.iter</code></td>
<td>
<p>Integer, the maximum number of iterations in the NIPALS
algorithm.</p>
</td></tr>
<tr valign="top"><td><code>tol</code></td>
<td>
<p>Positive real, the tolerance used in the NIPALS algorithm.</p>
</td></tr>
<tr valign="top"><td><code>logratio</code></td>
<td>
<p>one of ('none','CLR'). Specifies the log ratio
transformation to deal with compositional values that may arise from
specific normalisation in sequencing data. Default to 'none'</p>
</td></tr>
<tr valign="top"><td><code>multilevel</code></td>
<td>
<p>sample information for multilevel decomposition for
repeated measurements.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>scale= TRUE</code> is highly recommended as it will help obtaining orthogonal
sparse loading vectors.
</p>
<p><code>keepX</code> is the number of variables to select in each loading vector,
i.e. the number of variables with non zero coefficient in each loading
vector.
</p>
<p>Note that data can contain missing values only when <code>logratio = 'none'</code>
is used. In this case, <code>center=TRUE</code> should be used to center the data
in order to effectively ignore the missing values. This is the default
behaviour in <code>spca</code>.
</p>
<p>According to Filzmoser et al., a ILR log ratio transformation is more
appropriate for PCA with compositional data. Both CLR and ILR are valid.
</p>
<p>Logratio transform and multilevel analysis are performed sequentially as
internal pre-processing step, through <code><a href="../../mixOmics/help/logratio.transfo.html">logratio.transfo</a></code> and
<code><a href="../../mixOmics/help/withinVariation.html">withinVariation</a></code> respectively.
</p>
<p>Logratio can only be applied if the data do not contain any 0 value (for
count data, we thus advise the normalise raw data with a 1 offset). For ILR
transformation and additional offset might be needed.
</p>
<p>The principal components are not guaranteed to be orthogonal in sPCA. 
We adopt the approach of Shen and Huang 2008 (Section 2.3) to estimate 
the explained variance  in the case where the sparse loading vectors 
(and principal components) are not orthogonal. The data are projected 
onto the space spanned by the first loading vectors and the variance 
explained is then adjusted for potential correlation between PCs. 
Note that in practice, the loading vectors tend to be orthogonal if the
data are centered and scaled in sPCA.
</p>


<h3>Value</h3>

<p><code>spca</code> returns a list with class <code>"spca"</code> containing the
following components:
</p>

<dl>
<dt>ncomp</dt><dd><p>the number of components to keep in the
calculation.</p>
</dd> 
<dt>prop_expl_var</dt><dd><p>the adjusted percentage of variance
explained for each component.</p>
</dd> 
<dt>cum.var</dt><dd><p>the adjusted cumulative percentage of variances
explained.</p>
</dd>
<dt>keepX</dt><dd><p>the number of variables kept in each loading
vector.</p>
</dd> 
<dt>iter</dt><dd><p>the number of iterations needed to reach convergence
for each component.</p>
</dd> 
<dt>rotation</dt><dd><p>the matrix containing the sparse
loading vectors.</p>
</dd> 
<dt>x</dt><dd><p>the matrix containing the principal components.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Kim-Anh LÃª Cao, Fangzhou Yao, Leigh Coonan, Ignacio Gonzalez, Al J Abadi
</p>


<h3>References</h3>

<p>Shen, H. and Huang, J. Z. (2008). Sparse principal component
analysis via regularized low rank matrix approximation. <em>Journal of
Multivariate Analysis</em> <b>99</b>, 1015-1034.
</p>


<h3>See Also</h3>

<p><code><a href="../../mixOmics/help/pca.html">pca</a></code> and http://www.mixOmics.org for more details.
</p>


<h3>Examples</h3>

<pre>
data(liver.toxicity)
spca.rat &lt;- spca(liver.toxicity$gene, ncomp = 3, keepX = rep(50, 3))
spca.rat

## variable representation
plotVar(spca.rat, cex = 1)
## Not run: 
plotVar(spca.rat,style="3d")

## End(Not run)

## samples representation
plotIndiv(spca.rat, ind.names = liver.toxicity$treatment[, 3],
          group = as.numeric(liver.toxicity$treatment[, 3]))

## Not run: 
plotIndiv(spca.rat, cex = 0.01,
col = as.numeric(liver.toxicity$treatment[, 3]),style="3d")

## End(Not run)

## example with multilevel decomposition and CLR log ratio transformation
data("diverse.16S")
spca.res = spca(X = diverse.16S$data.TSS, ncomp = 5,
logratio = 'CLR', multilevel = diverse.16S$sample)
plot(spca.res)
plotIndiv(spca.res, ind.names = FALSE, group = diverse.16S$bodysite, title = '16S diverse data',
legend=TRUE)
</pre>

<hr /><div style="text-align: center;">[Package <em>mixOmics</em> version 6.18.1 <a href="00Index.html">Index</a>]</div>
</div></body></html>
